### ISSUE TITLE
Topic Saliency and Relevancy Frequency Distribution 

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#570]
**Describe the solution you'd like**
Adapt or extend the `FreqDist` visualizer to show not just the frequency of tokens in the corpus but rank them according to their relationship to a topic model (e.g. LDA). Ranked by *saliency*, the `FreqDist` will show the frequency of the terms that contribute the most information to all topics. Ranked by *relevance* the `FreqDist` will show the terms most relevant to a specific topic and the proportion of their frequency in the topic compared to the rest of the corpus. Metric definitions are:

1. **saliency(term w)** = frequency(w) * [sum_t p(t | w) * log(p(t | w)/p(t))] for topics t; see [Chuang et. al (2012)](http://vis.stanford.edu/files/2012-Termite-AVI.pdf)
2. **relevance(term w | topic t)** = λ * p(w | t) + (1 - λ) * p(w | t)/p(w); see [Sievert & Shirley (2014)](http://nlp.stanford.edu/events/illvi2014/papers/sievert-illvi2014.pdf)

This computation requires a model (so probably our best bet is to extend `FreqDist` to `TopicFreqDist` and make it a model visualizer). It also rquires a parameter, lambda. 

**Is your feature request related to a problem? Please describe.**
Implementing this will bring YB closer to being able to provide a PyLDAViz-like solution to topic modeling and clustering. 

**Examples**

Here is the `FreqDist` ranked by saliency to a topic model instead of ranked purely by freqency:

<img width="623" alt="screenshot 2018-08-21 06 47 13" src="https://user-images.githubusercontent.com/745966/44397360-1a0d3b00-a50e-11e8-94bc-e684dd743ff6.png">

Here is a topic's terms ranked by relevance and compared to the rest of the corpus (e.g. red vs. blue):

<img width="635" alt="screenshot 2018-08-21 06 48 56" src="https://user-images.githubusercontent.com/745966/44397420-4923ac80-a50e-11e8-9991-84452a30384f.png">

**Comment by rebeccabilbro** [comment]
I love this suggestion! For those interested, further reading:

- [Original R package: LDAvis](https://github.com/cpsievert/LDAvis)
- [Python port: pyLDAvis](https://pyldavis.readthedocs.io/en/latest/readme.html)
- [Latent Dirichlet Allocation](https://en.wikipedia.org/wiki/Latent_Dirichlet_allocation)
- [Topic Modeling](https://en.wikipedia.org/wiki/Topic_model)
- [sklearn.decomposition.LatentDirichletAllocation](http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.LatentDirichletAllocation.html)

![image](https://user-images.githubusercontent.com/8760385/44399530-f8b04d00-a515-11e8-89db-b260d85aeac7.png) 
(from Ch 6 of [Applied Text Analysis with Python](http://shop.oreilly.com/product/0636920052555.do))

**Comment by hminooei** [comment]
Hey guys,

Love the idea. Any progress on it? I want to use it for my current LDA model. :)

Thanks

**Comment by bbengfort** [comment]
Hi @hminooei thanks for using Yellowbrick! No progress on it, but we're certainly open for contributions and PRs!
